packages/generator/src/parser/parsed-service-metadata.ts

```diff
@@ -0,0 +1,19 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+
+import {
+  EdmxMetadataBaseExtended,
+  SwaggerMetadata
+} from './parser-types-common';
+import { EdmxMetadata as EdmxMetadataV2 } from './parser-types-v2';
+import { EdmxMetadata as EdmxMetadataV4 } from './parser-types-v4';
+
+export interface ParsedServiceMetadata<
+  T extends 'v2' | 'v4' | unknown = unknown
```

Comment:
could you please explain a bit regarding the use case for unknown?

Reply:
`unknown` really means unknown. We don't which it will be yet, only after parsing this is known.

Reply:
I removed that conditional type, as this is only relevant for two modules and can also be achieved by type assertions.